maven-artifact/src/main/java/org/apache/maven/artifact/handler/ArtifactHandler.java

```diff
@@ -19,7 +19,7 @@
 package org.apache.maven.artifact.handler;
 
 /**
- * An artifact handler defines for a dependency type, defined as Plexus role:<ul>
+ * An artifact handler provides metadata for an artifact derived from the dependency element that references the artifact:<ul>
  * <li>extension and classifier, to be able to download the file,</li>
```

Comment:
Does this always provide the info from the artifact's own pom.xml?

I'm a little confused because if the classifier is used to download the file as seen in the first bullet point then it has to come from the dependency element that loads the artifact, not the artifact's own pom.xml. Maybe the  first bullet point is wrong?

Reply:
See below, the artefacts own pom.xml is not used at all here afaik.

Reply:
on extension, there is no way to override it: then it's directly the extension
that's one reason why most people are confused by the (lack of) difference between type and extension, while there is a difference between the default classifier and the effective classifier